tools/importer-msgraph-metadata/internal/pipeline/task_translate_service.go

```diff
@@ -0,0 +1,181 @@
+
+	for _, resource := range resources {
+		// First scaffold all discovered services, version(s) and resources (categories)
+		if _, ok := sdkServices[resource.Service]; !ok {
+			sdkServices[resource.Service] = sdkModels.Service{
+				APIVersions:         make(map[string]sdkModels.APIVersion),
+				Generate:            true,
+				Name:                resource.Service,
+				ResourceProvider:    nil,
+				TerraformDefinition: nil,
+			}
+		}
+
+		if _, ok := sdkServices[resource.Service].APIVersions[resource.Version]; !ok {
+			sdkServices[resource.Service].APIVersions[resource.Version] = sdkModels.APIVersion{
+				APIVersion: resource.Version,
+				Generate:   true,
+				Preview:    versions.IsPreview(resource.Version),
+				Resources:  make(map[string]sdkModels.APIResource),
+				Source:     sdkModels.MicrosoftGraphMetaDataSourceDataOrigin,
+			}
+		}
+
+		if _, ok := sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category]; !ok {
+			sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category] = sdkModels.APIResource{
+				Constants:   make(map[string]sdkModels.SDKConstant),
+				Models:      make(map[string]sdkModels.SDKModel),
+				Operations:  make(map[string]sdkModels.SDKOperation),
+				ResourceIDs: make(map[string]sdkModels.ResourceID),
+			}
+		}
+
+		serviceModels := make(parser.Models)
+
+		// Populate everything else
+		for _, operation := range resource.Operations {
+			var resourceIdName *string
+
+			if operation.ResourceId != nil {
+				resourceIdName = &operation.ResourceId.Name
+
+				sdkResourceId, err := operation.ResourceId.DataApiSdkResourceId()
+				if err != nil {
+					return nil, err
+				}
+
+				sdkServices[resource.Service].APIVersions[resource.Version].Resources[resource.Category].ResourceIDs[operation.ResourceId.Name] = *sdkResourceId
+			}
+
+			var requestObject *sdkModels.SDKObjectDefinition
+			requestObjectIsCommonType := true
+
+			if operation.RequestModel != nil {
+				if !models.Found(*operation.RequestModel) {
+					return nil, fmt.Errorf("request model %q was not found for operation: %s", *operation.RequestModel, operation.Name)
+				}
+
+				if model := models[*operation.RequestModel]; !model.IsValid() {
+					return nil, fmt.Errorf("request model %q was invalid for operation: %s", *operation.RequestModel, operation.Name)
+				} else if !model.Common {
+					requestObjectIsCommonType = false
+
+					if err := serviceModels.MergeDependants(models, *operation.RequestModel); err != nil {
+						return nil, err
+					}
+				}
+
+				requestObject = &sdkModels.SDKObjectDefinition{
+					ReferenceName:             operation.RequestModel,
+					ReferenceNameIsCommonType: &requestObjectIsCommonType,
+					Type:                      sdkModels.ReferenceSDKObjectDefinitionType,
+				}
+			} else if operation.RequestType != nil {
+				requestObject = &sdkModels.SDKObjectDefinition{
+					Type: operation.RequestType.DataApiSdkObjectDefinitionType(),
```

Comment:
do we need to set `ReferenceNameIsCommonType: &requestObjectIsCommonType` here?

Reply:
We don't, as these will be builtin types (`string`, `bool` etc)